### DISCUSSION TITLE
Autocomplete `:` in the end of line such as `def`, `for`

### DISCUSSION THREAD

**Discussion by brettcannon** [microsoft/pylance-release#3790] [Enhancement]
### Discussed in https://github.com/microsoft/vscode-python/discussions/15606

<div type='discussions-op-text'>

<sup>Originally posted by **solzard** March  8, 2021</sup>
I always use PyCharm, Jetbrain's Python IDE, and try to make my newly-installed VSCode similar to that one.

VSCode is much lighter and faster, but one thing VSCode annoys me is it doesn't autocomplete `:` in the end of lines.
When writing `def` or `for` statement, PyCharm autocomplete `:` and moves to the indented next line using a shortcut called `complete current statement`. 
VSCode lacks `:`, so I have to go back and type it manually.

As I've googled so far, it seems there's no config or extension for this. 
Is such a feature supported in VSCode?

```python
# ←: current cursor before using the shortcut
def main(←)

# PyCharm
def main():
    [move to here]

# VSCode
def main()
    [move to here & lack a colon]
```</div>

**Comment by brettcannon** [comment]
Thanks for the feature request! We are going to give the community 60 days from when this issue was created to provide 5 👍 upvotes on the opening comment to gauge general interest in this idea. If there's enough upvotes then we will consider this feature request in our future planning. If there's unfortunately not enough upvotes then we will close this issue.

**Comment by brettcannon** [comment]
Thanks for the feature request! We are going to give the community 60 days from when this issue was created to provide 5 👍 upvotes on the opening comment to gauge general interest in this idea. If there's enough upvotes then we will consider this feature request in our future planning. If there's unfortunately not enough upvotes then we will close this issue.

**Comment by solzard** [comment]
Hi @brettcannon, your recent comment is exactly the same as the one you left two months ago. Did you paste a wrong boilerplate message? Or, will this issue repeat the 60 days period again?

**Comment by brettcannon** [comment]
@solzard sorry about that, wrong canned response 😅. What I meant to say was:

Thank you to everyone who upvoted this issue! Since the community showed interest in this feature request we will leave this issue open as something to consider implementing at some point in the future.

We do encourage people to continue 👍 this issue as it helps us prioritize our work based on what the community seems to want the most.

**Comment by brettcannon** [comment]
Moving to Pylance as a feature request.
